wagtail_newsletter/audiences.py

```diff
@@ -8,12 +9,14 @@
 from . import campaign_backends
 
 
-class CachedApiQueryish(Queryish):
+T = TypeVar("T", bound="AudienceBase")
+
+
+class CachedApiQueryish(Queryish, Generic[T]):
     cache_prefix: str
 
     @abstractmethod
-    def get_list(self):
-        raise NotImplementedError
+    def get_list(self) -> "dict[str, T]": ...
```

Comment:
I suppose the reason we use a "string type hint" here, rather than directly `-> dict[str, T]`, is because we still support Python 3.8 right?  

Given than this package is rather "cutting edge", and that event the security support of Python 3.8 will be over in 5 months, I wonder if we couldn't we drop support for that version? 

(That said, I guess we'll want to support what Wagtail's supported versions can run on themselves... which probably still includes 3.8 :slightly_smiling_face: )

Reply:
> I suppose the reason we use a "string type hint" here, rather than directly `-> dict[str, T]`, is because we still support Python 3.8 right?

Yep, that's right. And Wagtail still supports 3.8. You're right to point out that support for 3.8 is mostly academic at this point; I wouldn't mind dropping it TBH. On the other hand, it's not a big deal to enclose type hints in quotes, so I think I'll wait for a slightly stronger reason to change that, if you agree.

Reply:
Yup, it makes total sense to follow Wagtail's own Python versions support :+1: :slightly_smiling_face: